x-pack/plugins/observability_solution/infra/server/routes/infra/lib/utils.ts

```diff
@@ -27,23 +24,8 @@ const isValidFilter = (query: any): query is BoolQuery => {
     .every((clause) => Array.isArray(clause) || clause === undefined);
 };
 
-export const assertQueryStructure: (query: any) => asserts query is BoolQuery = (query) => {
-  if (!isValidFilter(query)) {
+export const assertQueryStructure: (query?: any) => asserts query is BoolQuery = (query) => {
+  if (!!query && !isValidFilter(query)) {
```

Comment:
Should we actually validate even if `!!query` is false? 
So we can have an `or` condition in this case to make sure that it's either existing or valid so something like:
```suggestion
  if (!!query || !isValidFilter(query)) {
```
Or (if there is no query then we don't need to validate): 
```suggestion
  if (!!query ?? !isValidFilter(query)) {
```

Reply:
> Or (if there is no query then we don't need to validate)
yeah, I think we can do this

Reply:
Well, actually, `isValidFilter` is only called if `query` is not `undefined`.

Reply:
If the query is `' '` or `{}` then it won't have `bool` so actually we can have a check in `isValidFilter` like 
`const boolClause = (query as estypes.QueryDslQueryContainer)?.bool;` instead because also `??` won't help here. Maybe it's an edge case or not possible because we have any and I assumed that it can be an empty object so we can also skip that in this case.